extension/src/webview/factory.ts

```diff
@@ -1,19 +1,20 @@
 import { Uri, ViewColumn, WebviewPanel, window } from 'vscode'
 import { BaseWebview } from '.'
 import { ViewKey, WebviewDetails } from './constants'
-import { WebviewState, WebviewData } from './contract'
 import { Resource } from '../resourceLocator'
 import { getWorkspaceRootUris } from '../vscode/workspaceFolders'
 
-export const isValidDvcRoot = (dvcRoot?: string): dvcRoot is string => !!dvcRoot
+const isValidDvcRoot = (dvcRoot?: string): dvcRoot is string => !!dvcRoot
+
+const needsDvcRoot = (viewKey: ViewKey): boolean =>
+  ![ViewKey.CONNECT, ViewKey.SETUP].includes(viewKey)
```

Comment:
[F] I think this is ok because the factory already knows about the type of webviews it is creating.